src/profile-logic/process-profile.js

```diff
@@ -1059,53 +1066,193 @@ export function processProfile(
+    // This will not remove the thread entry from the `threads` array right now
+    // and just replace it with a `null` value. We filter out the null values
+    // inside `serializeProfile` function.
+    return null;
+  }
+  // We need to update the stringTable. It's not possible with UniqueStringArray.
+  const stringArray = thread.stringTable.serializeToArray();
+
+  // We iterate all the markers and remove/change data depending on the PII
+  // status.
+  const markersToDelete = [];
+  if (
+    PIIToBeRemoved.shouldRemoveNetworkUrls ||
+    PIIToBeRemoved.shouldRemoveThreadsWithScreenshots.length > 0
+  ) {
+    for (let i = 0; i < thread.markers.length; i++) {
+      const currentMarker = thread.markers.data[i];
+
+      // Remove the all network URLs if user wants to remove them.
+      if (
+        PIIToBeRemoved.shouldRemoveNetworkUrls &&
+        currentMarker &&
+        currentMarker.type &&
+        currentMarker.type === 'Network'
+      ) {
+        // Remove the URI fields from marker payload.
+        // Mutating the payload here but it's safe because we copied the
+        // thread already.
+        removeNetworkMarkerURLs(currentMarker);
+
+        // Strip the URL from the marker name
+        const stringIndex = thread.markers.name[i];
+        stringArray[stringIndex] = stringArray[stringIndex].replace(/:.*/, '');
+      }
+
+      // Remove the screenshots if the current thread index is in the
+      // threadsWithScreenshots array
+      if (
+        PIIToBeRemoved.shouldRemoveThreadsWithScreenshots.includes(
+          threadIndex
+        ) &&
+        currentMarker &&
+        currentMarker.type &&
+        currentMarker.type === 'CompositorScreenshot'
+      ) {
+        const urlIndex = currentMarker.url;
+        // We are mutating the stringArray here but it's okay to mutate this
+        // here since we copied the string at the beginning while converting
+        // the string table to string array.
+        stringArray[urlIndex] = '';
+        markersToDelete.push(i);
+      }
+    }
+  }
+
+  // After iterating (or not iterating at all) the markers, if we have some
+  // markers we want to delete or user wants to delete the full time range,
+  // reconstruct the marker table and samples table without unwanted information.
+  if (
+    markersToDelete.length > 0 ||
+    PIIToBeRemoved.shouldFilterToCommittedRange !== null
+  ) {
+    // Filter marker table with given range and marker indexes array.
+    thread.markers = filterRawMarkerTableToRangeWithMarkersToDelete(
+      thread.markers,
+      markersToDelete,
+      PIIToBeRemoved.shouldFilterToCommittedRange
+    );
+
+    // While we are here, we are also filterig the thread samples
+    // to range.
+    if (PIIToBeRemoved.shouldFilterToCommittedRange !== null) {
+      const { start, end } = PIIToBeRemoved.shouldFilterToCommittedRange;
+      thread = (filterThreadSamplesToRange(thread, start, end): any);
```

Comment:
It doesn't look like we need an `any` here.

Reply:
Yeah, I did this initially because we were deleting `stringTable` field and adding `stringArray` instead(before splitting serialize and sanitize functions). Now we don't do that here and we can remove that `any` :)